zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -202,40 +226,78 @@ impl<F: FieldExt> Transaction<F> {
 
 #[derive(Debug, Default)]
 pub struct Call<F> {
+    /// The unique identifier of call in the whole proof, using the
+    /// `rw_counter` at the call step.
     pub id: usize,
+    /// Indicate if the call is the root call
     pub is_root: bool,
+    /// Indicate if the call is a create call
     pub is_create: bool,
+    /// The identifier of current executed bytecode
     pub opcode_source: F,
+    /// The `rw_counter` at the end of reversion of a call if it has
+    /// `is_persistent == false`
     pub rw_counter_end_of_reversion: usize,
+    /// The call index of caller
     pub caller_call_id: usize,
+    /// The depth in the call stack
     pub depth: usize,
+    /// The caller address
     pub caller_address: Address,
+    /// The callee address
     pub callee_address: Address,
+    /// The call data offset in the memory
     pub call_data_offset: usize,
+    /// The length of call data
     pub call_data_length: usize,
+    /// The return data offset in the memory
     pub return_data_offset: usize,
+    /// The length of return data
     pub return_data_length: usize,
+    /// The ether amount of the transaction
     pub value: Word,
+    /// TBD, Han will update this field
     pub result: Word,
+    /// Indicate if this call and all its caller have `is_success == true`
     pub is_persistent: bool,
+    /// Indicate if it's a static call
     pub is_static: bool,
 }
 
 #[derive(Clone, Debug, Default)]
 pub struct ExecStep {
-    pub call_idx: usize,
+    /// The index in the Transaction calls
+    pub call_index: usize,
+    /// The indices in the RW trace incurred in this step
     pub rw_indices: Vec<usize>,
+    /// The execution state for the step
     pub execution_state: ExecutionState,
+    /// The Read/Write counter before the step
     pub rw_counter: usize,
+    /// The program counter
     pub program_counter: u64,
+    /// The stack pointer
     pub stack_pointer: usize,
+    /// The amount of gas left
     pub gas_left: u64,
+    /// The gas cost in this step
     pub gas_cost: u64,
+    /// The memory size in bytes
     pub memory_size: u64,
+    /// The counter for state writes
     pub state_write_counter: usize,
+    /// The opcode corresponds to the step
     pub opcode: Option<OpcodeId>,
 }
 
+impl ExecStep {
+    pub fn memory_word_size(&self) -> u64 {
+        // The memory size must be multiple of words (32 bytes)
+        assert_eq!(self.memory_size % NUM_BYTES_WORD as u64, 0);
```

Comment:
@icemelon Does the `memory_size` have to be a multiple of 32 bytes? If we perform `MSTORE8`, does it result in a memory size that's not a multiple of 32 bytes?

Reply:
EVM always pads the memory to word size (see [here](https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L212-L216)). That's why `memory_size` has to be multiple of 32 bytes. I can elaborate it in the comment to address the confusion.

Reply:
The comment looks good to me